neqo-bin/src/udp.rs

```diff
@@ -26,10 +26,28 @@ pub struct Socket {
 impl Socket {
     /// Create a new [`Socket`] bound to the provided address, not managed externally.
     pub fn bind<A: std::net::ToSocketAddrs>(addr: A) -> Result<Self, io::Error> {
+        const ONE_MB: usize = 1 << 20;
         let socket = std::net::UdpSocket::bind(addr)?;
+        let state = quinn_udp::UdpSocketState::new((&socket).into())?;
+
+        let send_buf = state.send_buffer_size((&socket).into())?;
+        if send_buf < ONE_MB {
+            qdebug!("Increasing send buffer size from {send_buf} to {ONE_MB}");
+            state.set_send_buffer_size((&socket).into(), ONE_MB)?;
+        } else {
+            qdebug!("Default send buffer size is {send_buf}, not changing");
+        }
+
+        let recv_buf = state.recv_buffer_size((&socket).into())?;
+        if recv_buf < ONE_MB {
+            qdebug!("Increasing receive buffer size from {recv_buf} to {ONE_MB}");
+            state.set_recv_buffer_size((&socket).into(), ONE_MB)?;
+        } else {
+            qdebug!("Default receive buffer size is {recv_buf}, not changing");
+        }
 
```

Comment:
Should we add telemetry/stats about which sizes were actually in use for a connection?

Reply:
:+1: we can do that in `neqo_glue`.

Reply:
I'd like to keep the glue as minimal as possible. So adding this to `Stats` seems preferable.

Reply:
> I'd like to keep the glue as minimal as possible.

Agreed.

Note however that this would be the first time we set something in `neqo_transport::Stats` from outside of `neqo_transport`.